### PR TITLE
optimize compaction log

### PR DESCRIPTION
**Pull request by DeepThinker666**
optimize compaction log:
1. add error status message to log
2. add segment num/rows/disk size to compaction log

### CODE REVIEW

**Review by Astralidea — DISMISSED** [review]

File: be/src/storage/vectorized/compaction.cpp [hunk 1]
```diff
@@ -109,7 +109,11 @@ Status Compaction::do_compaction_impl() {
 
     LOG(INFO) << "succeed to do " << compaction_name() << ". tablet=" << _tablet->full_name()
               << ", output_version=" << _output_version.first << "-" << _output_version.second
-              << ", segments=" << segments_num << ". elapsed time=" << watch.get_elapse_second() << "s.";
+              << ", input infos [segments=" << segments_num << ", rows:" << _input_row_num
+              << ", disk size:" << _input_rowsets_size << "]"
+              << ", output infos [segments=" << _output_rowset->num_segments()
+              << ", rows:" << _output_rowset->num_rows() << ", disk size:" << _output_rowset->data_disk_size() << "]"
```
Comment (wyb) [inline be/src/storage/vectorized/compaction.cpp:115]: rows=xxx, disk size=xxx

Comment (DeepThinker666) [reply be/src/storage/vectorized/compaction.cpp]: done

**Review by wyb — DISMISSED** [review]

File: be/src/storage/vectorized/compaction.cpp [hunk 2]
```diff
@@ -195,7 +199,7 @@ Status Compaction::merge_rowsets(int64_t mem_limit, Statistics* stats_output) {
                 if (status.is_end_of_file()) {
                     break;
                 } else {
-                    return Status::InternalError("reader get_next error.");
+                    return Status::InternalError("reader get_next error:" + status.to_string());
```
Comment (imay) [inline be/src/storage/vectorized/compaction.cpp:202]: better to use fmt::format or strings::substitute

Comment (DeepThinker666) [reply be/src/storage/vectorized/compaction.cpp]: done

**Comment by DeepThinker666** [comment]
run starrocks_be_unittest

**Review by imay — APPROVED** [review]

**Pull request merged**